server/src/main/java/org/apache/iotdb/db/metadata/mtree/store/disk/schemafile/SchemaFile.java

```diff
@@ -115,13 +97,19 @@ public class SchemaFile implements ISchemaFile {
 
   private final Map<Integer, ISchemaPage> dirtyPages;
 
+  private ThreadLocal<int[]>
+      treeTrace; // a trace of b+tree index since no upward pointer within segments
```

Comment:
Why use ```ThreadLocal```? Shall we eliminate it by using ```Command Pattern```?

Reply:
Now there is no ```ThreadLocal``` then.